examples/larger-examples/datalog/ford-fulkerson.flix

```diff
@@ -0,0 +1,584 @@
+pub def main(): Unit \ IO =
+    let g1 = FordFulkerson.exampleGraph01() |> FordFulkerson.maxFlow(src = 0, dst = 5);
+    let g2 = FordFulkerson.exampleGraph02() |> FordFulkerson.maxFlow(src = 0, dst = 5);
+    let g3 = FordFulkerson.exampleGraph03() |> FordFulkerson.maxFlow(src = 0, dst = 3);
+    let g3 = FordFulkerson.exampleGraph04() |> FordFulkerson.maxFlow(src = 0, dst = 10);
+    println("Expected maximum flow 19, got ${g1}");
+    println("Expected maximum flow 20, got ${g2}");
+    println("Expected maximum flow 2000, got ${g3}");
+    println("Expected maximum flow 20, got ${g4}")
+
+mod FordFulkerson {
+
+    use Path.Path;
+
+    enum Path with ToString {
+        case Path(List[Int32])
+    }
+
+    instance Eq[Path] {
+        pub def eq(x: Path, y: Path): Bool =
+            let Path(xs) = x;
+            let Path(ys) = y;
+            List.length(xs) == List.length(ys)
+    }
+
+    instance Order[Path] {
+        pub def compare(x: Path, y: Path): Comparison =
+            let Path(xs) = x;
+            let Path(ys) = y;
+            if (List.length(xs) ==  List.length(ys))
+                Comparison.EqualTo
+            else if (List.length(xs) <= List.length(ys))
+                Comparison.GreaterThan
+            else
+                Comparison.LessThan
+    }
+
+    instance LowerBound[Path] {
+        // The longest list
+        pub def minValue(): Path = Path(List.range(1, 100))
+    }
+
+    instance PartialOrder[Path] {
+        pub def lessEqual(x: Path, y: Path): Bool =
+            let Path(xs) = x;
+            let Path(ys) = y;
+            List.length(xs) >= List.length(ys)
+    }
+
+    instance JoinLattice[Path] {
+        pub def leastUpperBound(x: Path, y: Path): Path =
+            let Path(xs) = x;
+            let Path(ys) = y;
+            if (List.length(xs) <= List.length(ys)) x else y
+    }
+
+    instance MeetLattice[Path] {
+        pub def greatestLowerBound(x: Path, y: Path): Path = ???
+    }
+
+
+    pub def init(y: Int32, x: Int32): Path =
+        Path(y :: x :: Nil)
+
+    pub def cons(z: Int32, p: Path): Path = match p {
+        case Path(xs) => Path(z :: xs)
+    }
+
+    type alias From = Int32
+    type alias To = Int32
+    type alias UpperCapacity = Int32
+    type alias Flow = Int32
+    type alias FlowNetwork = (From, To, UpperCapacity, Flow)
```

Comment:
Not sure how I feel about some of these.

The purpose is to be an example, not a real implementation for the library.

Reply:
Let me think about it.

Reply:
Okay, I think it significantly improves readability as everything is just `Int32` otherwise. :)

Reply:
But could also be polymorphic. Ill think on it.

Reply:
Keep the capacities as Int32 makes sense, but the nodes could be polymorphic.